### PR TITLE
Make a guess at imported podcast's enclosure prefix (if podtrac or blubrry) 

### LINKED ISSUE #276: On import, set enclosure prefix based on existing enclosure prefix
Could look for podtrac and blubrry in there at least, and craft a prefix based on those if they are present

### PR DESCRIPTION
**Pull request by gcampo88**
Fixes #276

### CODE REVIEW

File: app/models/podcast_import.rb [hunk 1]
```diff
@@ -148,6 +149,23 @@ def create_podcast
...
 
+  def enclosure_prefix(podcast_item)
+    prefix = ''
+    link = [podcast_item.feedburner_orig_enclosure_link,
+            podcast_item.enclosure.try(:url),
+            podcast_item.media_contents.first.try(:url)].find do |url|
+              url.try(:match, /podtrac/) || url.try(:match, /blubrry/)
+            end
+    if link.try(:match, /podtrac/)
+      podtrac_end = link.index('mp3') + 'mp3/'.length
```
Comment (gcampo88) [inline app/models/podcast_import.rb]: Respectfully, Mr. Hound, I think this is clearer

File: test/models/podcast_import_test.rb [hunk 2]
```diff
@@ -92,6 +92,38 @@
...
+                                                 'redirect.mp3/media.blubrry.com/'
+
+      item.feedburner_orig_enclosure_link = nil
+      item.enclosure.url = sample_link1
+      importer.enclosure_prefix(item).must_equal 'https://www.podtrac.com/pts/redirect.mp3/'
+
+      item.feedburner_orig_enclosure_link = 'something_without_those_words'
+      item.enclosure.url = sample_link2
+      importer.enclosure_prefix(item).must_equal 'http://www.podtrac.com/pts/redirect' +
+                                                 '.mp3/media.blubrry.com/'
```
Comment (kookster) [inline test/models/podcast_import_test.rb]: I believe this is missing `99percentinvisible/`
Blubrry prefix includes an identifier for the podcast after `media.blubrry.com`

e.g. it's my understanding that the blubrry prefix for 99pi is: `media.blubrry.com/99percentinvisible`, not just `media.blubrry.com`

File: app/models/podcast_import.rb [hunk 3]
```diff
@@ -148,6 +149,23 @@ def create_podcast
...
+      prefix += link[0...podtrac_end]
+    end
+    if link.try(:match, /blubrry/) && !prefix.match(/blubrry/)
+      prefix += 'media.blubrry.com/'
```
Comment (kookster) [inline app/models/podcast_import.rb]: I think you also need the identifier for the podcast, that is the path immediately after `media.blubrry.com/`

Comment (kookster) [inline app/models/podcast_import.rb]: https://create.blubrry.com/resources/podcast-media-download-statistics/instructions-for-developers/

File: test/models/podcast_import_test.rb (hunk 2 above)
Comment (kookster) [inline test/models/podcast_import_test.rb]: https://create.blubrry.com/resources/podcast-media-download-statistics/getting-started/

Comment (gcampo88) [inline test/models/podcast_import_test.rb]: oops! will amend!

File: app/models/podcast_import.rb [hunk 4]
```diff
@@ -148,6 +149,26 @@ def create_podcast
...
+            podcast_item.media_contents.first.try(:url)].find do |url|
+              url.try(:match, /podtrac/) || url.try(:match, /blubrry/)
+            end
+    if link.try(:match, /podtrac/)
+      podtrac_end = link.index('mp3') + 'mp3/'.length
+      prefix += link[0...podtrac_end]
+    end
+    if link.try(:match, /blubrry/) && !prefix.match(/blubrry/)
+      blubrry_start = link.index('media.blubrry.com/')
+      blubrry_end = blubrry_start + 'media.blubrry.com/'.length
```
Comment (gcampo88) [inline app/models/podcast_import.rb]: overruled!

**Comment by gcampo88** [comment]
back to you @kookster

File: app/models/podcast_import.rb [hunk 5]
```diff
@@ -148,6 +149,26 @@ def create_podcast
     podcast
   end
 
+  def enclosure_prefix(podcast_item)
+    prefix = ''
+    link = [podcast_item.feedburner_orig_enclosure_link,
```
Comment (kookster) [inline app/models/podcast_import.rb]: these matches work, but there is this regex geek in me that feels the need to say you can do both the match to see if there is blubrry, in there and get out the prefix info in one regex:
```
> l = "http://www.podtrac.com/pts/redirect.mp3/media.blubrry.com/99percentinvisible/dovetail.prxu.org/99pi/9350e921-b910-4b1c-bbc6-2912d79d014f/248-Atom-in-the-Garden-of-Eden.mp3"

# if there is a match, the blubrry prefix will be in match data `1`
> bp = l.match(/\/(media\.blubrry\.com\/[^\/]+)\//)
=> #<MatchData "/media.blubrry.com/99percentinvisible/" 1:"media.blubrry.com/99percentinvisible">
```

likewise for podtrac, you can get the podtrac specific subdomain and path:
```
> pp = l.match(/\/(\w+\.podtrac.com\/.+?\.mp3)/)
=> #<MatchData "/www.podtrac.com/pts/redirect.mp3" 1:"www.podtrac.com/pts/redirect.mp3">
```

Comment (kookster) [inline app/models/podcast_import.rb]: your way totally works, but as we old perl hackers like to say [TMTOWTDI](https://en.wikipedia.org/wiki/There's_more_than_one_way_to_do_it)

**Comment by gcampo88** [comment]
back to you @kookster !

**Pull request merged**